assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -91,6 +104,110 @@ function MediaProvider({ children }) {
     ]
   );
 
+  const resetMedia = useCallback(
+    ({ pagingNum: p = 1 } = {}) => {
+      fetchMediaStart({ pagingNum: p });
+      getMedia({ mediaType, searchTerm, pagingNum: p })
+        .then(({ data }) => {
+          setMedia({
+            media: data,
+          });
+        })
+        .catch(fetchMediaError);
+    },
+    [
+      setMedia,
+      fetchMediaError,
+      fetchMediaStart,
+      getMedia,
+      mediaType,
+      searchTerm,
+    ]
+  );
+
+  const uploadMediaFromLibrary = useCallback(
+    async (files) => {
+      try {
+        const localMedia = await Promise.all(
+          files.map(getResourceFromLocalFile).reverse()
+        );
+        const filesUploading = files.map((file) => uploadFile(file));
+        setMedia({
+          media: [...localMedia, ...media],
```

Comment:
The same problem remains. The `localMedia` files are in "resource" shape, while `media` are in "attachment" shape. We could consider always converting to "resource" shape whenever `setMedia()` is called.